packages/core/src/codecs/octetstream-codec.ts

```diff
@@ -59,38 +59,18 @@ export default class OctetstreamCodec implements ContentCodec {
             throw new Error("Lengths do not match, required: " + parameters.length + " provided: " + bytes.length);
         }
 
-        let dataType = undefined;
         let dataLength = bytes.length;
-
-        // check @type property for further information
-        if(schema['@type'] !== undefined) {
-            let semTypes: Array<string> = schema["@type"];
-            // check for endianness semantic type
-            // see http://www.meta-share.org/ontologies/meta-share/meta-share-ontology.owl/documentation/index-en.html
-            let endianSem = semTypes.find(v => v.endsWith(":bigEndian") || v.endsWith(':littleEndian'));
-            bigendian = endianSem === undefined ? bigendian : endianSem.endsWith(":bigEndian");
-            // check for numeric datatype semantic
-            // see https://www.w3.org/TR/xmlschema-2/
-            let typeSem = semTypes.find(v => /:(unsigned)?(short|int|long|float|double|byte)/.test(v.toLowerCase()));
-            if(typeSem) {
-                // check for sign semantic type
-                signed = typeSem.toLowerCase().indexOf('unsigned') === -1;
-                dataType = /(short|int|long|float|double|byte)/.exec(typeSem.toLowerCase())[1];
-            }
-        }
+        let dataType : string = schema.type;
```

Comment:
Here I'm assuming the schema is a string that it isn't neither a NumericSchema nor an IntegerSchema. I don't quite like that. Should I try to define a new schema inside td-tools package?

Reply:
Yeah, it would be better. We may need to change it in the future cause is not yet introduced in the TD. I'd add a TODO comment here explaining the situation.